### DISCUSSION TITLE
Problems with markup rendering

### DISCUSSION THREAD

**Discussion by JLucasGaldino** [nvim-neorg/neorg#1312] [Q&A]
Hello everyone,

This is my first time posting an issue/discussion on GitHub, so please forgive me if this has already been answered.

# The problem
Bold, italics, links, and strike-through elements do not render at all. There is no difference from the normal text.

# Solutions I've tried

## Inspecting (:inspect) these elements. This is what I get:
@neorg.markup.bold.norg links to @markup
@neorg.markup.italic.norg links to @markup
@neorg.links.description.norg links to @markup
@neorg.markup.strikethrough.norg links to @markup

## Check the highlight settings of these elements (:hi). This is what I get:
@neorg.markup.bold xxx links to @markup.strong
@neorg.markup.italic xxx links to @markup.italic
@neorg.links.description xxx links to @markup.link.url
@neorg.markup.strikethrough xxx links to @markup.strikethrough

## Verify how these settings are applying in md files, where text formatting seems to be working well:
@text.strong.markdown_inline links to @text.strong
@text.emphasis.markdown_inline links to @text.emphasis
@text.uri.markdown_inline links to @text.uri
@text.strike.markdown_inline links to @text.strike

## Try different concealer settings
By setting concealer=2, the special symbols disappear and the links are properly hidden, but there is still no change in how the markup renders.

# My setup
OS: Ubuntu 22.04
Editor: Nvim v0.9.4
Major config: NvChad

Thanks so much for the help!

**Comment by ALVAROPING1** [comment]
recently, upstream nvim-treesitter moved from `@text.*` highlight capture groups to `@markup.*`. Neorg has followed this change, so you need to update your colorscheme to use these capture groups (or configure neorg to link its groups to different ones)

**Comment by faustozamparelli** [comment]
I have the same problem, i am using the default nvim theme and a custom iterm2 colorscheme, headings render properly in neorg but underline, strikethrough, bold, italic... doesn't. I have tried everything. The only solution seems to be installing tokyonight theme but i like my colorscheme and don't want to change it and keep my terminal colorscheme in nvim. What can i do? i understand the @markup.* but i have no idea how to change it. @ALVAROPING1  could you help?

**Reply by ALVAROPING1** [reply]
adding this to your neovim config somewhere where it gets executed after loading your colorscheme should restore the functionality of the old names by linking them the new ones to them (should handle all names, not just those used by neorg):

```lua
vim.api.nvim_set_hl(0, "@variable.parameter", { link = "@parameter" })
vim.api.nvim_set_hl(0, "@variable.member", { link = "@field" })
vim.api.nvim_set_hl(0, "@module", { link = "@namespace" })
vim.api.nvim_set_hl(0, "@number.float", { link = "@float" })
vim.api.nvim_set_hl(0, "@string.special.symbol", { link = "@symbol" })
vim.api.nvim_set_hl(0, "@string.regexp", { link = "@string.regex" })
vim.api.nvim_set_hl(0, "@function.method", { link = "@method" })
vim.api.nvim_set_hl(0, "@function.method.call", { link = "@method.call" })
vim.api.nvim_set_hl(0, "@keyword.import", { link = "@include" })
vim.api.nvim_set_hl(0, "@keyword.storage", { link = "@storageclass" })
vim.api.nvim_set_hl(0, "@keyword.repeat", { link = "@repeat" })
vim.api.nvim_set_hl(0, "@keyword.debug", { link = "@debug" })
vim.api.nvim_set_hl(0, "@keyword.exception", { link = "@exception" })
vim.api.nvim_set_hl(0, "@keyword.conditional", { link = "@conditional" })
vim.api.nvim_set_hl(0, "@keyword.conditional.ternary", { link = "@conditional.ternary" })
vim.api.nvim_set_hl(0, "@keyword.directive", { link = "@preproc" })
vim.api.nvim_set_hl(0, "@keyword.directive.define", { link = "@define" })
vim.api.nvim_set_hl(0, "@markup.strong", { link = "@text.strong" })
vim.api.nvim_set_hl(0, "@markup.italic", { link = "@text.emphasis" })
vim.api.nvim_set_hl(0, "@markup.strikethrough", { link = "@text.strike" })
vim.api.nvim_set_hl(0, "@markup.underline", { link = "@text.underline" })
vim.api.nvim_set_hl(0, "@markup.heading", { link = "@text.title" })
vim.api.nvim_set_hl(0, "@markup.raw", { link = "@text.literal" })
vim.api.nvim_set_hl(0, "@markup.raw.block", { link = "@text.literal.block" })
vim.api.nvim_set_hl(0, "@markup.quote", { link = "@text.quote" })
vim.api.nvim_set_hl(0, "@markup.math", { link = "@text.math" })
vim.api.nvim_set_hl(0, "@markup.environment", { link = "@text.environment" })
vim.api.nvim_set_hl(0, "@markup.environment.name", { link = "@text.environment.name" })
vim.api.nvim_set_hl(0, "@markup.link", { link = "@text.reference" })
vim.api.nvim_set_hl(0, "@markup.link.url", { link = "@text.uri" })
vim.api.nvim_set_hl(0, "@markup.link.label", { link = "@string.special" })
vim.api.nvim_set_hl(0, "@markup.list", { link = "@punctuation.special" })
vim.api.nvim_set_hl(0, "@comment.error", { link = "@text.danger" })
vim.api.nvim_set_hl(0, "@comment.warning", { link = "@text.warning" })
vim.api.nvim_set_hl(0, "@comment.note", { link = "@text.note" })
vim.api.nvim_set_hl(0, "@comment.todo", { link = "@text.todo" })
vim.api.nvim_set_hl(0, "@diff.plus", { link = "@text.diff.add" })
vim.api.nvim_set_hl(0, "@diff.minus", { link = "@text.diff.delete" })
```

Once neovim 0.10 is released, you might want to try removing this patch since it might be unnecessary (default colorschemes likely will support the new names directly, which is better)

**Comment by pysan3** [comment]
For those who don't know what the **** is going on, here's an explanation on Discord.

> let me explain. approx 2 weeks ago a PR to neovim master was added to make all queries align with the upstream tree-sitter. so all `@text -> @markup.text` and so on.
> this also affected the highlight group names, and most colorscheme plugins updated accordingly, or if not, there's a remap in neovim to make old colorschemes working (like the ones that dont support ts colors alltogether `Normal -> @markup.text`, not to keep `@text` working).
> neorg also changed all names to `@markup.*` hardcoded, so noone make the existing `@text` etc to work. 
> so it's nobody's fault, everyone is doing their job for their good heart but it doesn't get combined well when you are on an older version that that tree-sitter PR.
> ~~[this](https://discord.com/channels/834325286664929280/834326555160608769/1210132892156567582)~~ [_what @ALVAROPING1 posted above_] is a workaround you can do yourself until 0.10 gets its first stable release.
> please read the conversation that starts from here for more info https://discord.com/channels/834325286664929280/834326555160608769/1210117779408297985, no kidding, youll see how much we suffered to identify the problem lol

TLDR; upgrade to nightly is the easiest way to fix. If you can't or don't want to, use @ALVAROPING1 's snippet.

**Comment by JLucasGaldino** [comment]
Sorry for the slow answer everyone! And thanks for the help. I am going to add my own solution here, which is a minimal version of @ALVAROPING1 's, plus some custom conceal settings. I am going to close this, since the issue seems to be solved for everyone.

```
-- These groups allow customization of the Neorg highlights
vim.api.nvim_set_hl(0, '@neorg.markup.bold.norg', { link = 'NeorgCustomBold'})
vim.api.nvim_set_hl(0, '@neorg.markup.italic.norg', { link = 'NeorgCustomItalic'})
vim.api.nvim_set_hl(0, '@neorg.markup.strikethrough.norg', { link = 'NeorgCustomStrikethrough'})
vim.api.nvim_set_hl(0, '@neorg.links.description.norg', { link = 'NeorgCustomLink'})

-- This autocommand allows for automatic concealing of Neorg files.
-- There is a plugin already that does that, but this is a failsafe.
vim.api.nvim_create_autocmd({"BufEnter", "BufWinEnter"}, {
  pattern = {"*.norg"},
  callback = function ()
    vim.opt.conceallevel = 2
    vim.opt.concealcursor = 'nc'
  end
})
```

**Reply by Rac0r** [reply]
Hey I am currently having the same issue!
I use NvChad as a base config aswell and everything with neorg works except the markdown rendering.
In which config file did you put these lines?

**Reply by pillzu** [reply]
I am running into a similar issue. 

Upgrading to nightly is not really an option for me and I am not sure how to get the markdown rendering to work using @ALVAROPING1 suggestion (I am not sure where to place it, but i tried to do so `vim.cmd.colorscheme` but it had no specific effect except for the highlighting to vanish as well. 

Note that somehow bold works, but none of the other text decorations work (strike-through, italics etc).

Would appreciate any support regarding the same.
